mindmeld/active_learning/classifiers.py

```diff
@@ -0,0 +1,418 @@
+                ending indexes of intents with the given domain.
+        """
+        eval_stats = defaultdict(dict)
+        eval_stats["num_sampled"] = len(data_bucket.sampled_queries)
+        preds_single = self.train_single(data_bucket, eval_stats)
+        return_preds_multi = isinstance(heuristic, MULTI_MODEL_HEURISTICS)
+        preds_multi = self.train_multi(data_bucket) if return_preds_multi else None
+        domain_indices = (
+            self.domain_indices if isinstance(heuristic, KLDivergenceSampling) else None
+        )
+        return eval_stats, preds_single, preds_multi, domain_indices
+
+    def train_single(self, data_bucket: DataBucket, eval_stats: defaultdict = None):
+        """Trains a single model to get a 2D probability array for single-model selection strategies.
+        Args:
+            data_bucket (DataBucket): Databucket for current iteration
+            eval_stats (defaultdict): Evaluation metrics to be included in accuracies.json
+        Returns:
+            preds_single (List): 2D array with probability vectors for unsampled queries
+        """
+        return self._train_single(
+            sampled_queries=data_bucket.sampled_queries,
+            unsampled_queries=data_bucket.unsampled_queries,
+            test_queries=data_bucket.test_queries,
+            label_map=data_bucket.label_map,
+            eval_stats=eval_stats,
+        )
+
+    def _train_single(
+        self,
+        sampled_queries: List[ProcessedQuery],
+        unsampled_queries: List[ProcessedQuery],
+        test_queries: List[ProcessedQuery],
+        label_map: LabelMap,
+        eval_stats: Dict = None,
+    ):
+        """Helper function to train a single model and obtain a 2D probability array.
+        Args:
+            sampled_queries List[ProcessedQuery]: Current set of sampled queries in DataBucket.
+            unsampled_queries List[ProcessedQuery]: Current set of unsampled queries in DataBucket.
+            test_queries List[ProcessedQuery]: Set of test queries in DataBucket.
+            label_map LabelMap: Class that stores index mappings for a MindMeld app.
+                (Eg. Domain2Id, Intent2Id)
+            eval_stats (Dict): Evaluation metrics to be included in accuracies.json
+        Returns:
+            preds_single (List): 2D array with probability vectors for unsampled queries
+        """
+        # Domain_Level
+        dc_queries_prob_vectors, dc_eval_test = self.domain_classifier_fit_eval(
+            sampled_queries=sampled_queries,
+            unsampled_queries=unsampled_queries,
+            test_queries=test_queries,
+            domain2id=label_map.domain2id,
+        )
+        if eval_stats:
+            MindMeldALClassifier._update_eval_stats_domain_level(
+                eval_stats, dc_eval_test
+            )
+        preds_single = dc_queries_prob_vectors
+
+        # Intent_Level
+        if self.training_level == TRAIN_LEVEL_INTENT:
```

Comment:
If `training_level` is set to intent, can't we just skip `domain_classifier_fit_eval()`?

Reply:
Currently, we are training the domain classifier to obtain domain-level accuracy metrics

Reply:
It seems that `preds_single ` is getting overwritten if `TRAIN_LEVEL_INTENT `. If thats the case, why cant we have an `if else` clause so that we dont run `domain_classifier_fit_eval ` when `TRAIN_LEVEL_INTENT`